pcap.pyx

```diff
@@ -369,22 +373,23 @@ cdef class pcap:
 
     def __next__(self):
         cdef pcap_pkthdr *hdr
-        cdef char *pkt
+        cdef u_char *pkt
         cdef int n
         while 1:
-            Py_BEGIN_ALLOW_THREADS
-            n = pcap_ex_next(self.__pcap, &hdr, &pkt)
-            Py_END_ALLOW_THREADS
+            with nogil:
+                n = pcap_ex_next(self.__pcap, &hdr, &pkt)
             if n == 1:
-                return (hdr.ts.tv_sec + (hdr.ts.tv_usec / 1000000.0),
-                        PyBuffer_FromMemory(pkt, hdr.caplen))
+                return (
+                    hdr.ts.tv_sec + (hdr.ts.tv_usec / 1000000.0),
+                    get_buffer(pkt, hdr.caplen),
+                )
             elif n == 0:
-                return None
+                continue
```

Comment:
Is it possible that this leads to an infinite loop? Is this related to this issue: https://github.com/pynetwork/pypcap/issues/49?